thingsboard_gateway/gateway/tb_gateway_service.py

```diff
@@ -984,8 +985,14 @@ def __convert_telemetry_to_ts(data):
                 if isinstance(item['ts'], int):
                     telemetry_with_ts.append({"ts": item["ts"], "values": {**item["values"]}})
                 else:
-                    log.warning('Data has invalid TS (timestamp) format! Using generated TS instead.')
-                    telemetry_with_ts.append({"ts": int(time() * 1000), "values": {**item["values"]}})
+                    try:
+                        dt_obj = datetime.datetime.strptime(item["ts"], '%Y-%m-%dT%H:%M:%SZ').replace(tzinfo=datetime.timezone.utc)
+                        timestamp = int(dt_obj.timestamp() * 1000)
+                        telemetry_with_ts.append({"ts": timestamp, "values": {**item["values"]}})
+                        log.warning('Data has string TS (timestamp) format! Convert to integer.')
+                    except:
```

Comment:
After adding a dataFormat as a separated parameter - we can check is it present and if yes - use converting for incoming ts.